ats/pkb/pkb.py

```diff
@@ -44,18 +45,64 @@ def process_relations(node: nodes.ASTNode):
                 if isinstance(parent, nodes.StmtNode):
                     parents[node.__stmt_id] = parent.__stmt_id
 
+            if isinstance(
+                node, (nodes.StmtNode, nodes.StmtLstNode, nodes.ProcedureNode)
+            ):
+                uses_stack.append(node)
+                uses_stack.append([])
```

Comment:
?

Reply:
Po włożeniu na stos powyższych nodów dodaję na stos pustą listę która będzie zapamiętywać zmienne których używają. Przykład : Po włożeniu noda StmtList dfs idzie w głąb pierwszego dziecka i wkłada na stos jego elementy a gdy do dojdzie na sam dół zdejmuje je ze stosu i dodaje zmienne do listy z których używa pierwsze dziecko następnie wchodzi w głąb drugiego dziecka i robi to samo itd.

Reply:
ja sie pytam czemu dajecie node albo liste. Nie wiecie co to typy?

Reply:
Można zrobić to bez list, ale zrobiłem z listami dla ułatwienia. Różnica jest taka że bez list na bieżąco będą dopisywane do słownika zmienne dla konkretnych statementów, a z listami zbieram do listy wszystkie zmienne dla statementa i jak już mam wszystkie to wtedy dodaje je do słownika pomijając powtórzenia.